### PR TITLE
feat(swap): startExchange accept NG types

### PR DESCRIPTION
**Pull request by sprohaszka-ledger**
### 📝 Description

* Add startExchange conformance to NG ExchangeTypes
* Add types on startExchange LLD method calls
* Check for app version in case of NG ExchangeType call

### ❓ Context

- **JIRA or GitHub link**: [LIVE-10434](https://ledgerhq.atlassian.net/browse/LIVE-10434)

### ✅ Checklist

Pull Requests must pass the CI and be code reviewed. Set as Draft if the PR is not ready.

- [ ] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - StartExchange (swap) on LLD

---

### 🧐 Checklist for the PR Reviewers

- [ ] **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- [ ] **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- [ ] **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- [ ] **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- [ ] **Any new dependencies** have been justified and documented.
- [ ] **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-10434]: https://ledgerhq.atlassian.net/browse/LIVE-10434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx [hunk 1]
```diff
@@ -30,6 +30,13 @@ export type Data = {
   amountExpectedTo?: number;
 };
 
+export function isCompleteExchangeData(data: unknown): data is Data {
+  if (data === null || typeof data !== "object") {
+    return false;
+  }
+  return "signature" in data && "binaryPayload" in data ? true : false;
```
Comment (CremaFR) [inline apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx:37]:
```suggestion
return "signature" in data && "binaryPayload" in data;
```

It should work without checking if true is true?

File: apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/StartExchange/index.tsx [hunk 2]
```diff
@@ -4,15 +4,30 @@ import Box from "~/renderer/components/Box";
...
+export type Data = {
+  onCancel?: (error: Error) => void;
+  exchangeType: ExchangeType;
+  onResult: (startExchangeResult: string) => void;
+};
+export function isStartExchangeData(data: unknown): data is Data {
+  if (data === null || typeof data !== "object") {
+    return false;
+  }
+  return "exchangeType" in data ? true : false;
```
Comment (CremaFR) [inline apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/StartExchange/index.tsx:19]:
```suggestion
  return "exchangeType" in data;
```

same here  in returns already a boolean right?

**Review by CremaFR — DISMISSED** [review]
LGTM

File: apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx (hunk 1 above)
Comment (sprohaszka-ledger) [reply apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx:37]: you're right 🤦

**Review by CremaFR — APPROVED** [review]

**Pull request merged**